test/Microsoft.Azure.WebJobs.Host.UnitTests/Singleton/SingletonEnd2EndTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using System;
+using System.Threading;
+using Xunit;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Host.UnitTests.Singleton
+{
+    public class SingletonEnd2EndTests 
```

Comment:
There is already a SingletonEndToEndTests test suite. Let's add any new tests to that one